services/multitenant/tenantstats.go

```diff
@@ -105,37 +99,18 @@ func NewStats(routerDBs map[string]jobsdb.MultiTenantJobsDB) *Stats {
 	t := Stats{}
 	config.RegisterDurationConfigVariable(60, &t.jobdDBQueryRequestTimeout, true, time.Second, []string{"JobsDB.Multitenant.QueryRequestTimeout", "JobsDB.QueryRequestTimeout"}...)
 	config.RegisterIntConfigVariable(3, &t.jobdDBMaxRetries, true, 1, []string{"JobsDB." + "Router." + "MaxRetries", "JobsDB." + "MaxRetries"}...)
+	t.RouterDBs = routerDBs
+	t.reset()
```

Comment:
**_Note:_** it is (unlikely but still) possible to use a non-started `tenantstats.Stats`, more specifically, router's `backendConfigSubscriber` calls `UpdateWorkspaceLatencyMap` whenever a new workspace is detected.

For that reason, all struct's internal maps need to be initialized during `tenantstats.NewStats`